server/src/domain/system-config/dto/system-config-reverse-geocoding.dto.ts

```diff
@@ -9,4 +36,14 @@ export class SystemConfigReverseGeocodingDto {
   @IsEnum(CitiesFile)
   @ApiProperty({ enum: CitiesFile, enumName: 'CitiesFile' })
   citiesFileOverride!: CitiesFile;
+
+  @IsBoolean()
+  @ValidateIf(isEnabled)
+  useMapbox!: boolean;
```

Comment:
I would like to add Google maps in the future. Can we change this to an enums? So like `provider: ReverseGeocodingProvider`

The options can be reverse local geocoder or map box.

The validate if can be if provider is mailbox, then require the access key. In the UI it can be a dropdown select instead.

For lots of providers, you will probably need a key, so we should rename this to just apiKey. If provider is mailbox you can still do a custom validation to check it is valid. If provider is Google maps we can also do a validation of it. I don't think we need a separate field for maps API key vs map box API key.